builder_test.go

```diff
@@ -411,6 +572,7 @@ func testBuilder(t *testing.T, when spec.G, it spec.S) {
 		when("platform api < 0.4", func() {
 			it.Before(func() {
 				builder.PlatformAPI = api.MustParse("0.3")
+				config.PlatformAPI = "0.3"
```

Comment:
If we follow the suggestion in https://github.com/buildpacks/lifecycle/pull/520/files#r568171284 we won't need to modify the config that gets passed to each buildpack.

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/520/commits/845f01c70137cd8ddf8c399a1b6a93f3199d6b72.